.gitlab-ci.yml

```diff
@@ -20,6 +20,7 @@ variables:
   timeout: 20m
   before_script:
     - pyenv global 3.12 3.7 3.8 3.9 3.10 3.11 3.13-dev
+    - export _CI_DD_AGENT_URL=http://${HOST_IP}:8126/
```

Comment:
To clarify: the reason we need to add this new `_CI_DD_AGENT_URL` var is because with GitLab, we (currently) don't have a way to spin up a separate service/test agent, but we did with CircleCI?

Reply:
In CircleCI, we were using agentless mode, so we didn't care what the agent URL was, but I decided it would be better to use an agent in GitLab since a- it's there, and b- it means we don't have to get an API key.

The `CIVisibility` tracer needs "its own" agent URL because otherwise it'll try to send traces to the test agent, in the snapshot test jobs (and the existing use of `_DD_CIVISIBILITY_USE_CI_CONTEXT_PROVIDER` already forces it to create a different tracer).